mne/source_estimate.py

```diff
@@ -1696,6 +1771,127 @@ def get_peak(self, tmin=None, tmax=None, mode='abs',
+                 tstep=None, subject=None, verbose=None):
+        SourceEstimate.__init__(self, data, vertices=vertices, tmin=tmin,
+                                tstep=tstep, subject=subject, verbose=verbose)
+
+    @verbose
+    def save(self, fname, ftype='h5', verbose=None):
+        """Save the full source estimate to an HDF5 file.
+
+        Parameters
+        ----------
+        fname : string
+            The file name to write the source estimate to, should end in
+            '-stc.h5'.
+        ftype : string
+            File format to use. Currently, the only allowed values is "h5".
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose).
+            Defaults to self.verbose.
+        """
+        if ftype != 'h5':
+            raise ValueError('VectorSourceEstimate objects can only be '
+                             'written as HDF5 files.')
+
+        if not fname.endswith('.h5'):
+            fname += '-stc.h5'
+
+        write_hdf5(fname,
+                   dict(vertices=self.vertices, data=self.data, tmin=self.tmin,
+                        tstep=self.tstep, subject=self.subject),
+                   title='mnepython', overwrite=True)
+
+        logger.info('[done]')
+
+    def __abs__(self):
+        """Compute magnitude of activity without directionality.
+
+        Returns
+        -------
+        stc : instance of SourceEstimate
+            The source estimate without directionality information.
+        """
+        data_abs = np.linalg.norm(self.data, axis=1)
+        return SourceEstimate(data_abs, self.vertices, self.tmin, self.tstep,
+                              self.subject, self.verbose)
+
+    def normal(self, src):
+        """Compute activity orthogonal to the cortex.
+
+        Parameters
+        ----------
+        src : instance of SourceSpaces
+            The source space for which this source estimate is specified.
+
+        Returns
+        -------
+        stc : instance of SourceEstimate
+            The source estimate only retaining the activity orthogonal to the
+            cortex.
+        """
+        normals = np.vstack([s['nn'][v] for s, v in zip(src, self.vertices)])
+        data_norm = np.einsum('ijk,ij->ik', self.data, normals)
+        return SourceEstimate(data_norm, self.vertices, self.tmin, self.tstep,
+                              self.subject, self.verbose)
+
+    @copy_function_doc_to_method_doc(plot_vec_source_estimates)
+    def plot(self, time, subject=None, surface='white', hemi='lh',
+             colormap='hot', time_label='time=%0.3f s', transparent=None,
+             vector_alpha=1.0, brain_alpha=0.4,
+             scale_factor=None, subjects_dir=None, figure=None, views='lat',
+             colorbar=True, clim='auto', size=800, background='white',
+             foreground='black'):
+
+        return plot_vec_source_estimates(self, time, subject=subject,
```

Comment:
Have you considered allowing `plot_source_estimates` to take `SourceEstimate` or `VectorSourceEstimate` as an input? It seems like it might help deduplication?